controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -1297,6 +1292,17 @@ String getScopedStreamName(String scope, String stream) {
     // region reader group
     abstract ReaderGroup newReaderGroup(final String scope, final String name);
 
+    /* This is a data class that represents a writer mark. Writers send mark information
+     * for watermarking purposes, containing time and position.
+     */
+    @Data
+    @AllArgsConstructor
+    public static class TxnWriterMark {
```

Comment:
we should make this a top level class instead of a nested class. 
its used in the Store interface as well. So keeping this in the implementation  of store class does not look good.

Reply:
This is just a "helper" class used for passing Transactional WriterMarks & used only by the CommitRequestHandler to pass data to the Store, as such I think it should be an inner class. We do have another class `io.pravega.controller.store.stream.records.WriterMark` to represent the actual WriterMark.

Reply:
but this is used in the store interface in one of the methods, right? if so, this should not be defined in the abstract implementation of the store interface where this is used.

Reply:
Made TxnWriterMark a top level class, as suggested.